File: Lab0/suma.c
/** @file suma.c
 *  Programa que suma
 */
#include <stdio.h>
#include <stdlib.h>

/**
 * Esta función toma un arreglo de números reales y devuelve la suma de todos sus elementos.
 * @param size tamaño del arreglo.
 * @param array arreglo de numeros reales.
 * @return resultado de la suma.
 */
double sumar(int size, double* array)
{
	double result = 0;
	int i;
	for (i = 0; i < size; i++) {
		result += array[i];
	}
	return result;
}

int main(int argc, char **argv)
{
	double array[argc - 1];
	int i;
	for (i = 1; i < argc; i++) {
		array[i-1] = atof(argv[i]);
	}
	
	printf("Programa en C\nEl resultados es %f\n", sumar(argc - 1, array));
	return 0;
}

File: Lab0/suma.py
#!/usr/bin/python
## @package suma
#  Script para sumar.
#
import sys
import math

## Esta función suma los elementos de una lista.
#  @param data la lista.
#  @return el resultado de la suma.
def sumar(data):
    result = 0.0;
    for i in data:
        result = result + i
    return result


if __name__ == "__main__":
    datos = [];
    for i in range(1, len(sys.argv)):
        datos.append(float(sys.argv[i]))

    print "Programa en Python"
    print "El resultado es: %f" % sumar(datos)


